datashape/coretypes.py

```diff
@@ -1031,6 +1035,38 @@ def __getitem__(self, key):
     def __str__(self):
         return pprint(self)
 
+category_predicates = [
+    (lambda x: 0 < x <= 1 << 8, np.uint8),
+    (lambda x: 1 << 8 < x <= 1 << 16, np.uint16),
+    (lambda x: 1 << 16 < x <= 1 << 32, np.uint32),
+    (lambda x: 1 << 32 < x <= 1 << 64, np.uint64)
+]
+
+
+def dtype_for_length(n):
+    if n <= 0:
+        raise ValueError('number of categories must be greater than 0')
+    for f, dtype in category_predicates:
+        if f(n):
+            return dtype
+    raise ValueError('Number of categories is greater than 2 ** 64')
+
+
+class Categorical(Mono):
+    """Unordered categorical type.
+    """
+
+    __slots__ = 'type', 'categories'
+    cls = MEASURE
+
+    def __init__(self, categories, type=None):
+        self.categories = tuple(categories)
+        self.type = type or datashape.discover(self.categories).measure
+
+    def __repr__(self):
+        return '%s[%s]' % (type(self).__name__.lower(),
+                           pformat(self.categories))
```

Comment:
What would you suggest here? I can limit the length that's printed with `pformat`